controlplane/kubeadm/controllers/controller_test.go

```diff
@@ -83,6 +83,55 @@ var _ = Describe("Kubeadm Control Plane Controller", func() {
 				return err
 			}, 10*time.Second).Should(HaveOccurred())
 		})
+
+		It("should update the observed generation after a reconcile", func(t *testing.T) {
```

Comment:
I don't think this should be using `t *testing.T` here:

```suggestion
		It("should update the observed generation after a reconcile", func() {
```

Below, should be using the global `Expect` rather than the `NewWithT`